src/bindgen/language_backend/java_jna.rs

```diff
@@ -0,0 +1,860 @@
+                .iter()
+                .enumerate()
+                .map(|(index, arg)| IndexedFunctionArg {
+                    name: &arg.name,
+                    ty: &arg.ty,
+                    index,
+                })
+                .collect::<Vec<_>>(),
+        );
+
+        out.write(");");
+    }
+
+    fn write_type<W: Write>(&mut self, out: &mut SourceWriter<W>, t: &Type) {
+        match t {
+            Type::Ptr { ty, .. } => match &**ty {
+                Type::Ptr { .. } => out.write("PointerByReference"),
+                Type::Path(path) => {
+                    write!(out, "{}ByReference", path.export_name())
+                }
+                Type::Primitive(primitive) => {
+                    let typ = match primitive {
+                        PrimitiveType::Void => "Pointer",
+                        PrimitiveType::Bool => "Pointer",
+                        PrimitiveType::Char => "ByteByReference",
+                        PrimitiveType::SChar => "ByteByReference",
+                        PrimitiveType::UChar => "ByteByReference",
+                        PrimitiveType::Char32 => "Pointer",
+                        PrimitiveType::Float => "FloatByReference",
+                        PrimitiveType::Double => "DoubleByReference",
+                        PrimitiveType::VaList => "PointerByReference",
+                        PrimitiveType::PtrDiffT => "PointerByReference",
+                        PrimitiveType::Integer { kind, .. } => {
+                            match kind {
+                                IntKind::Short => "ShortByReference",
+                                IntKind::Int => "IntByReference",
+                                IntKind::Long => "NativeLongByReference",
+                                IntKind::LongLong => "LongByReference",
+                                IntKind::SizeT => "NativeLongByReference", // TODO probably not right
+                                IntKind::Size => "NativeLongByReference", // TODO probably not right
+                                IntKind::B8 => "ByteByReference",
+                                IntKind::B16 => "ShortByReference",
+                                IntKind::B32 => "IntByReference",
+                                IntKind::B64 => "LongByReference",
+                            }
+                        }
+                    };
+                    write!(out, "{typ}")
+                }
+                Type::Array(_, _) => out.write("Pointer"),
+                Type::FuncPtr { .. } => out.write("CallbackReference"),
+            },
+
+            Type::Path(path) => {
+                write!(out, "{}", path.export_name())
+            }
+            Type::Primitive(primitive) => {
+                //https://github.com/java-native-access/jna/blob/master/www/Mappings.md
+                let typ = match primitive {
+                    PrimitiveType::Void => "void",
+                    PrimitiveType::Bool => "boolean",
```

Comment:
Not right, this is 1 byte in rust, 4 in java.

Reply:
fixed, now using a dedicated JNA IntegerType with two constants TRUE and FALSE